packages/salesforcedx-utils-vscode/src/util/authInfo.ts

```diff
@@ -146,24 +146,11 @@ export class OrgAuthInfo {
     });
   }
 
-  public static async getOrgApiVersion(): Promise<string | undefined> {
-    const connection = await WorkspaceContext.getInstance().getConnection();
-    const apiVersion = connection.getApiVersion();
-    return apiVersion ? String(apiVersion) : undefined;
-  }
-
-  public static async getUserId(): Promise<string | undefined> {
-    const connection = await WorkspaceContext.getInstance().getConnection();
-    const userId = connection.getAuthInfoFields().userId ?? (await connection.identity()).user_id;
-    return userId;
-  }
-
-  public static async getAuthFields(): Promise<AuthFields> {
-    const connection = await WorkspaceContext.getInstance().getConnection();
+  /** Get auth fields for a connection */
+  public static async getAuthFields(connection: Connection): Promise<AuthFields> {
```

Comment:
<img width="526" height="147" alt="Image" src="https://github.com/user-attachments/assets/8791ff08-38ec-4fb7-8eef-9a8f477a7e36" />

```suggestion
  public static getAuthFields(connection: Connection): Promise<AuthFields> {
```